llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -8997,12 +9012,14 @@ bool llvm::matchSimpleRecurrence(const PHINode *P, BinaryOperator *&BO,
     switch (Opcode) {
     default:
       continue;
-    // TODO: Expand list -- xor, div, gep, uaddo, etc..
+    // TODO: Expand list -- sdiv, srem, fadd etc.
```

Comment:
why remove the `xor` and `gep` and `uaddo` todos

Reply:
- I can't think of a possible recurrence with xor.
- GEP isn't a BinaryOperator, is it? What possible recurrences are possible?
- I can't find uaddo in the LangRef, and don't know what it is.

Reply:
> * I can't think of a possible recurrence with xor.
> 
Isn't that also true for `sdiv`/`srem`?

>     * GEP isn't a BinaryOperator, is it? What possible recurrences are possible?
> 
If it matches `ptradd` probably.

>     * I can't find uaddo in the LangRef, and don't know what it is.

Probably uadd_sat.